src/gpio.rs

```diff
@@ -311,19 +331,36 @@ macro_rules! gpio {
                         let mode = 0b11;
                         let bits = (cnf << 2) | mode;
 
+                        let mut res = $PXi { _mode: PhantomData };
+
+                        match initial_state {
+                            State::High => res.set_high(),
+                            State::Low  => res.set_low(),
+                        }
+
                         cr
                             .cr()
                             .modify(|r, w| unsafe {
                                 w.bits((r.bits() & !(0b1111 << offset)) | (bits << offset))
                             });
 
-                        $PXi { _mode: PhantomData }
+                        res
                     }
-
-                    /// Configures the pin to operate as an push pull output pin
+                    /// Configures the pin to operate as an push-pull output pin.
+                    /// Initial state will be low.
```

Comment:
Do we want to explicitly state the state of the pin in this function?

Also, it looks like the changes to this file aren't explicitly used in the PR, were they added by accident?

Reply:
Well now i'm confused, these changes were part of #5 which is already merged, why are they showing up as diffs for this PR?